pkg/controller/statusmanager/status_manager.go

```diff
@@ -83,6 +84,7 @@ func (status *StatusManager) Set(reachedAvailableLevel bool, conditions ...condi
 	for i := 0; i < conditionsUpdateRetries; i++ {
 		err := status.set(reachedAvailableLevel, conditions...)
 		if err == nil {
+			monitoring.SetReadyGauge(reachedAvailableLevel)
```

Comment:
is there somewhere in the code where we need to set `SetReadyGauge(false)` (for example after it turned Ready, or if its going to be undeployed gracefully if there is such scenario) ?

Reply:
not sure what you mean.. My premise is that if statusManager works well to describe the availability of CNAO, then we can just follow that.

Reply:
just in the case of err != nil, do we need to update the Ready to 0 ? or to some value representing unknown ?
or just keep the old value

Reply:
if we failed to set the status, then the last status remained, so I think we should also keep the metric the same.. 

>(are there other cases of tear down where we wont reach here but do want to set it to 0 ?)

not that I can think of.